ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx

```diff
@@ -46,58 +48,99 @@ function EmbeddedTable({
                         Deployment
                     </Th>
+                    <Th width={20} sort={getSortParams('Cluster')}>
+                        Cluster
+                    </Th>
                     <Th width={30} sort={getSortParams('Namespace')}>
                         Namespace
                     </Th>
-                    <Th width={30} sort={getSortParams('Cluster')}>
-                        Cluster
-                    </Th>
+                    <Th>Count</Th>
                 </Tr>
             </Thead>
             {deployments.map(({ id, name, namespace, cluster, listeningEndpoints }, rowIndex) => {
-                const isExpanded = expandedRowSet.has(id);
+                // A row is expanded if
+                //   - the "are all rows expanded" toggle is on and the row is not in the toggled set
+                //   - the "are all rows expanded" toggle is off and the row is in the toggled set
+                const isExpanded = areAllRowsExpanded
+                    ? !invertedExpansionRowSet.has(id)
+                    : invertedExpansionRowSet.has(id);
+                const count = listeningEndpoints.length;
                 return (
                     <Tbody key={id} isExpanded={isExpanded}>
                         <Tr>
-                            {listeningEndpoints.length > 0 ? (
-                                <Td
-                                    expand={{
-                                        rowIndex,
-                                        isExpanded,
-                                        onToggle: () => expandedRowSet.toggle(id),
-                                    }}
-                                />
-                            ) : (
-                                <Td />
-                            )}
-                            <Td dataLabel="Deployment">{name}</Td>
-                            <Td dataLabel="Namespace">{namespace}</Td>
+                            <Td
+                                expand={{
+                                    rowIndex,
+                                    isExpanded,
+                                    onToggle: () => invertedExpansionRowSet.toggle(id),
+                                }}
+                            />
+                            <Td dataLabel="Deployment">
+                                <Button
+                                    variant={ButtonVariant.link}
+                                    isInline
+                                    component={LinkShim}
+                                    href={`${riskBasePath}/${id}`}
+                                >
+                                    {name}
+                                </Button>
+                            </Td>
                             <Td dataLabel="Cluster">{cluster}</Td>
+                            <Td dataLabel="Namespace">{namespace}</Td>
+                            <Td dataLabel="Listening endpoints count">{count}</Td>
```

Comment:
Aha, move in column brings to mind something I missed in the initial iteration: it is a common convention to right align a column of counts. Not a blocker to merge.